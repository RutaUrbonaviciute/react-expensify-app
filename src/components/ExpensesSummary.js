import React from 'react';
import numeral from 'numeral';
import { connect } from 'react-redux';
import getExpensesTotal from '../selectors/expense-total';
import selectExpenses from '../selectors/expenses';

export const ExpensesSummary = ({ expenseCount, expensesTotal }) => {
  const makePlural = expenseCount > 1 ? 's' : ''

  return (
    <div>
      <h1>
        Viewing {expenseCount} expense{makePlural} totalling {numeral(expensesTotal / 100).format('$0,0.00')}
      </h1>
    </div>
  )
}

const mapStateToProps = (state) => {
  const expenses = selectExpenses(state.expenses, state.filters)
  return {
    expenseCount: expenses.length,
    expensesTotal: getExpensesTotal(expenses)
  }
}
export default connect(mapStateToProps)(ExpensesSummary)