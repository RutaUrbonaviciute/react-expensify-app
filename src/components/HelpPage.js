import React from 'react';

const HelpPage = () => (
  <div>
    this is help page
  </div>
)

export default HelpPage