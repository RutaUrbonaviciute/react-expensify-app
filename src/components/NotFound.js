import React from 'react';
import { Link } from 'react-router-dom';

const NotFound = () => (
  <div>
    404 not found bithezz - <Link to="/">go fkin home</Link>
  </div>
)

export default NotFound