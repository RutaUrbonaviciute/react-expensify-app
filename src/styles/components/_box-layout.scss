.box-layout {
  align-items: center;
  background: url('/images/bg4.jpg');
  background-size: cover;
  display: flex;
  height: 100vh;
  justify-content: center;
  width: 100vw;

  &__box {
    background: fade-out($light-green, .20);
    border-radius: 3px;
    padding: $l-size $m-size;
    text-align: center;
    width: 25rem;
  }

  &__title {
    margin: 0 0 $m-size 0;
    line-height: 1;
    color: $dark-green;
  }
}