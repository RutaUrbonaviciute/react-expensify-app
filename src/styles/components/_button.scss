.button {
  $this: &;

  background-color: $orange;
  border-radius: 3px;
  border: none;
  color: #fefefe;
  display: inline-block;
  font-size:$font-size-large;
  font-weight: 300;
  line-height: 1;
  padding: $m-size;
  text-decoration: none;

  &:hover, &:focus {
    background-color: $dark-orange;
    transition: background-color .3s ease-out;
  }
}

.button--link {
  background: none;


  &:hover, &:focus {
    background: none;
    color: $orange;
    transition: color .3s ease-out;
  }
}

.button-secondary {
  background-color: $light-green;
  color: $body;

  &:hover, &:focus {
    background-color: $dark-green;
    color:  #fefefe;
    transition: background-color .3s ease-out, color .3s ease-out;
  }
}