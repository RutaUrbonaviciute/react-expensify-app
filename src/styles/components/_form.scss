.form {
  display: flex;
  flex-direction: column;

  > * {
    margin-bottom: $m-size;
  }

  &__error {
    color: red;
    font-style: italic;
    margin: 0 0 $m-size 0;

  }
}