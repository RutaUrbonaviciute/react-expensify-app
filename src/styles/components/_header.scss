.header {
  background-color: $dark-green;

&__content {
  align-items: center;
  display: flex;
  justify-content: space-between;
  padding: $s-size 0;
}

  &__title {
    color: #fefefe;
    text-decoration: none;
     h1 {
       margin: 0
     }
  }
}