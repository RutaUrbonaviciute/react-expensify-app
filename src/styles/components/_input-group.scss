.input-group {
  flex-direction: column;
  display: flex;
  margin-bottom: $m-size;

  &__item {
    margin-bottom: $s-size;
  }
}


@media (min-width: $desktop-breakpoint) {
  .input-group {
    align-items: center;
    flex-direction: row;
    margin-bottom: $l-size;

    &__item {
      margin: 0 $m-size 0 0;
    }
  }
}