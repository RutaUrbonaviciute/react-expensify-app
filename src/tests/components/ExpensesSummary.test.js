import React from 'react';
import { shallow } from 'enzyme';
import { ExpensesSummary } from '../../components/ExpensesSummary';

test('should render ExpensesSummary correctly with 1 expense', () => {
  const wrapper = shallow(<ExpensesSummary expenseCount={1} expensesTotal={235} />)
  expect(wrapper).toMatchSnapshot()
})

test('should render ExpensesSummary correctly with multiple expenses', () => {
  const wrapper = shallow(<ExpensesSummary expenseCount={2} expensesTotal={34535} />)
  expect(wrapper).toMatchSnapshot()
})