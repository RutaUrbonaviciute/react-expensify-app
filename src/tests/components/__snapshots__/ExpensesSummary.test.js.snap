// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render ExpensesSummary correctly with 1 expense 1`] = `
<div>
  <h1>
    Viewing 
    1
     expense
     totalling 
    $2.35
  </h1>
</div>
`;

exports[`should render ExpensesSummary correctly with multiple expenses 1`] = `
<div>
  <h1>
    Viewing 
    2
     expense
    s
     totalling 
    $345.35
  </h1>
</div>
`;
